core/src/main/java/com/oath/oak/MemoryManager.java

```diff
@@ -1,56 +1,23 @@
-/*
- * Copyright 2018 Oath Inc.
- * Licensed under the terms of the Apache 2.0 license.
- * Please see LICENSE file in the project root for terms.
- */
-
 package com.oath.oak;
 
-import com.oath.oak.NativeAllocator.OakNativeMemoryAllocator;
+import java.io.Closeable;
 import java.nio.ByteBuffer;
 
+public interface MemoryManager extends Closeable {
 
-public class MemoryManager {
-    private final OakMemoryAllocator keysMemoryAllocator;
-    private final OakMemoryAllocator valuesMemoryAllocator;
-
-    public MemoryManager(OakMemoryAllocator memoryAllocator) {
-        assert memoryAllocator != null;
-
-        this.valuesMemoryAllocator = memoryAllocator;
-        this.keysMemoryAllocator = memoryAllocator;
+    enum Allocate {
+        KEY, VALUE;
     }
 
-    public void close() {
-        valuesMemoryAllocator.close();
-        keysMemoryAllocator.close();
-    }
+    boolean isClosed();
 
-    // how many memory is allocated for this OakMap
-    public long allocated() {
-        return valuesMemoryAllocator.allocated();
-    }
+    long allocated();
 
-    // allocateSlice is used when the blockID (of the block from which the ByteBuffer is allocated)
```

Comment:
Please bring back those or other comments explaining the interface...

Reply:
These comments do not add any information. We only allocate slices, and we use slices for both keys and values.